bulk_indexer.go

```diff
@@ -269,14 +269,15 @@ func (b *BulkIndexer) BytesUncompressedFlushed() int {
 }
 
 type BulkIndexerItem struct {
-	Index      string
-	DocumentID string
-	Body       io.WriterTo
+	Index            string
+	DocumentID       string
+	Body             io.WriterTo
+	DynamicTemplates map[string]string
```

Comment:
@axw I was initially a little hesitant to pass in a map here, as Body is a io.WriterTo containing serialized bytes which makes it look inconsistent. But now I'm fine with the idea, as DynamicTemplates does not really require specific data structures and is only a `map[string]string`. WDYT?

Reply:
I think of it like this:
 - `DynamicTemplates` has a known structure, which fits naturally to `map[string]string`
 - `Body` just needs to be JSON, but we shouldn't be too prescriptive about how the JSON encoding is done

So I think it's perfectly fine to use `map[string]string` here.